### PR TITLE
Remove non-documented legacy API parts (fix #1715, fix #1721)

### LINKED ISSUE #1715: Figure out whether the legacy API parts behind auth are still used (DELETE / POST / PUT)
Tracker: #1680.

The legacy API can do authentication and supports  `POST`, `PUT`, and `DELETE` operations on add-ons and versions. This is undocumented however. Is still still used ? This issue is about finding out the answer to that question :-)

### LINKED ISSUE #1721: Figure out if all other GET endpoints are still useful in the legacy API
Tracker: #1680 

The legacy API exposes a GET API to fetch versions attached to an add-on and a few other things in the "2" api version. We should list those endpoints and figure out if they are still useful.

### PR DESCRIPTION
**Pull request by diox**
Investigations that led to this commit are in #1715 and #1721.

Everything behind authentication in the old API is removed in this commit. In addition, everything behind api version "2" is also removed as it is undocumented. (Firefox uses version "1.5", the
newer stuff we build uses version "3". AFAIK version "2" was only used for things like the add-on builder, which no longer exists)

This also removes piston, since it was only useful for those undocumented APIs.

This does *not* switch the remaining parts of the legacy API to DRF yet.

### CODE REVIEW

File: src/olympia/api/authentication.py [hunk 1]
```diff
@@ -1,90 +0,0 @@
...
-                'RestOAuth' in getattr(request._request, 'authed_from', [])):
-            request.user = request._request.user
-            return request.user, None
-
-    def authenticate_header(self, request):
-        return 'OAuth realm="%s"' % self.www_authenticate_realm
-
-
-class AMOOAuthAuthentication(OAuthAuthentication):
-    """^^^MOO!!!  Adds user to the request object."""
```
Comment (diox) [inline src/olympia/api/authentication.py]: :cow:

**Comment by EnTeQuAk** [comment]
r+ generally, this is so damn awesome.

If we're removing piston with this, let's do this also in the `INSTALLED_APPS` and everywhere else if we can, or is it still being used for some other stuff?

**Comment by diox** [comment]
It's removed from `INSTALLED_APPS` in `src/olympia/lib/settings_base.py`

**Comment by diox** [comment]
I filed #1728 to remove the piston-related database tables later once we know it's safe to do so.

**Pull request merged**